src/commands/publish.ts

```diff
@@ -185,19 +186,25 @@ async function printRevisionSummary(
 ): Promise<void> {
   const artifacts = await artifactProvider.listArtifactsForRevision(revision);
   if (artifacts.length > 0) {
-    const artifactData = artifacts.map(ar => [
-      ar.filename,
-      formatSize(ar.storedFile.size),
-      ar.storedFile.lastUpdated || '',
-    ]);
+    const artifactData = artifacts.map(ar => {
+      const contentType = detectContentType(ar.filename);
```

Comment:
To clarify, this content type guessing was initially meant to be done for GCS target only.

Artifact providers technically can store content types for files as part of metadata, but honestly, I don't really see a consistent way of showing this information: that content type we're showing, was it guessed? Is it part of artifact metadata? Also, we don't really attach the contentType we detected to the artifact itself, so later in the pipeline (e.g. inside GCS target) we have to detect it again.

Reply:
> To clarify, this content type guessing was initially meant to be done for GCS target only.

I see. Okay, so currently it's part of the underlying `gcsApi.uploadArtifact()` method, meaning it applies to the GCS artifact provider as well. Do you anticipate that causing problems, such that I should extract it and stick it back into the target only?

In any case, I take your point that at this point in the process (when we're just logging the list of files) we don't actually record content type on the artifact , so if the detected type differs from the stored type, and we're targeting something other than GCS, this could give a false impression of what we're uploading elsewhere. Two options I see here:

1) Take out the detection at this stage and just report the real data we have (attached to the artifact object) on content type. In conjunction, make it more obvious when, as part of GCS target upload, we change the type.

2) Take out this reporting all together.

I lean towards 1) since I feel like, within reason, more information is better than less information. WDYT?

Reply:
I'm fine with 1), but let's keep in mind that metadata provided by GCS artifact provider will also be "guessed" by GCS client in the "upload" command (if we don't do anything smart there, I mean).

Reply:
Sorry - does this mean you think I should take it out of `gcsApi` and put it only in `gcsTarget`, or not? (Separate from the logging.)

Reply:
"it" -- you mean `detectContentType`? It can stay in `files`

Reply:
No, sorry - I don't mean where the code for the method lives. I mean, currently we're doing the detection-and-content-type-setting in the generic GCS upload method, so any `.js` or `.map` file which gets uploaded either by the GCS artifact provider or by the GCS target has its content type set manually. My question is, should this only happen for the target, and not the artifact provider?

Reply:
Ah, hm, good question 🤔 
Let's leave the detection where it is for now, and we can separately discuss if we want to allow setting custom content types from command line as part  of `craft artifact upload`, because it's related.

Reply:
> if we want to allow setting custom content types from command line as part of craft artifact upload

Oh, good point. I actually did have that in my mental spec for the upload command (since it's something zeus can do and which I've seen used by SDKs when they upload to zeus), but hadn't gotten as far as thinking about how that and this question would interact. Will add a note there about this and we can revisit at that point.

Reply:
Okay, this turns out to be more complicated than I realized. Took the detection out of the listing and left a TODO for the next round.